articles/contributing/style-guide.md

```diff
@@ -0,0 +1,157 @@
+
+Similarly, _agent nouns_ can be used to construct function and UDT names from operation names, as in the case of the name `Encoder` for a UDT that is strongly associated with `Encode`. 
+
+### Shorthand ###
+
+The above advice notwithstanding, there are many forms of shorthand that see common and pervasive use in quantum computing.
+We suggest using existing and common shorthand where it exists, especially for operations that are intrinsic to the operation of a target machine.
+For example, we choose the name `X` instead of `ApplyX`, and `Rz` instead of `RotateAboutZ`.
+
+### Proper Nouns in Names ###
+
+
+
+- Avoid using people's names in operation and function names where reasonable.
+  Consider using names that describe the implemented functionality;
+  e.g. `CCNOT` versus `Toffoli` or `CSWAP` versus `Fredkin `.
+  In sample code, consider using names that are familiar to the community reading each particular example, even if that would otherwise run counter to these suggestions.
+  **NB:** names should still appear in documentation comments.
+
+
+
+- If an operation or function is not intended for direct use, but rather should be used by a matching callable which acts by partial application, consider using a name ending with `Impl` for the callable that is partially applied.
+  By contrast, if an operation or function should never be directly called by a user, consider indicating this with a leading `_`.
+- Value argument names should be descriptive; avoid one or two letter names where possible.
+- Generic argument names should be single capital letters where the role of a type is obvious.
+  Otherwise, consider using a short capitalized word prefaced by `T` (e.g.: `TOutput`).
+- Denote types in argument and variable names where it is ambiguous, but omit where it is clear from context.
+  Type names should be suffixes (`targetQubit`) where reasonable.
+- Denote scalar types by their literal names (`flagQubit`), and array types by a plural (`measResults`).
+  For arrays of qubits in particular, consider denoting such types by `Register` where the name refers to a sequence of qubits that are closely related in some way.
+- If several operations or functions are related by the functor variants supported by their arguments, denote this by suffixes `A`, `C` or `CA` to their names.
+- Where reasonable, arrays should have names that are pluralized (e.g.: `things`).
+- Variables used as indices into arrays should begin with `idx` and should be singular (e.g.: `things[idxThing]`).
+  In particular, strongly avoid using single-letter variable names as indices; consider using `idx` at a minimum.
+- Variables used to hold lengths of arrays should begin with `n` and should be pluralized (e.g.: `nThings`).
+
+## Argument Conventions ##
+
+The argument ordering conventions here largely derive from thinking of partial application as a generalization of currying 𝑓(𝑥, 𝑦) ≡ 𝑓(𝑥)(𝑦).
```

Comment:
Should this be written in LaTeX rather than in italic symbols? We had reports of italics not rendering on some systems - I think this might be the case in a browser as well.